crates/tests/src/sync.rs

```diff
@@ -0,0 +1,227 @@
+        let sealed_hash = sealed_genesis.hash();
+        assert_eq!(sealed_hash, genesis_hash, "genesis hashes should match, we computed {sealed_hash:?} but geth computed {genesis_hash:?}");
+
+        // === create many blocks ===
+
+        let nonces = 0..10000u64;
+        let txs = nonces
+            .map(|nonce| {
+                // create a tx that just sends to the zero addr
+                TypedTransaction::Eip1559(Eip1559TransactionRequest::new()
+                    .to(H160::zero())
+                    .value(1u64)
+                    .nonce(nonce))
+            });
+
+        for tx in txs {
+            // send the tx - geth will mine a block with just this transaction in it if we await
+            // here rather than joining concurrent sends
+            provider.send_transaction(tx, None).await.unwrap();
+        }
+
+        // wait for a certain number of blocks to be mined
+        let block = provider.get_block_number().await.unwrap();
+        println!("block num after creating transactions: {block}");
+        assert!(block > U64::zero());
+
+        // get the tip so we can send it to reth
+        let tip_block = provider.get_block(BlockNumber::Latest).await.unwrap().unwrap();
+        let tip_hash = tip_block.hash.unwrap().0.into();
+
+        // === initialize reth networking stack ===
+
+        let secret_key = SecretKey::new(&mut rand::thread_rng());
+        let (reth_p2p, reth_disc) = unused_tcp_udp();
+
+        // may not need to set up a hello as we should be advertising compatible capabilities and
+        // protocol versions anyways
+        // TODO: it's sort of redundant setting BOTH the status and the genesis hash, since they
+        // both contain the genesis hash. a discrepancy is probably an error. could we enforce this
+        // somethow?
+        let config = NetworkConfig::builder(Arc::new(NoopProvider::default()), secret_key)
+            .listener_addr(reth_p2p)
+            .discovery_addr(reth_disc)
+            .genesis_hash(sealed_genesis.hash())
+            .status(status)
+            .build();
+
+        let network = NetworkManager::new(config).await.unwrap();
+        let handle = network.handle().clone();
+
+        // convert ethers genesis to chainspec
+        let chainspec = genesis_to_chainspec(&genesis);
+
+        // initialize db
+        let reth_temp_dir = tempfile::tempdir().expect("should be able to create reth data dir");
+        let db = Arc::new(init_db(reth_temp_dir.path()).unwrap());
+
+        // initialize consensus
+        let consensus = Arc::new(BeaconConsensus::new(chainspec.consensus));
+
+        // build reth and start the pipeline
+        let reth: RethTestInstance<Env<WriteMap>> = RethBuilder::new()
+            .db(db)
+            .consensus(consensus)
+            .genesis(chainspec.genesis)
+            .network(handle.clone())
+            .tip(tip_hash)
+            .build();
```

Comment:
I like that we're testing syncing. Would have also liked some way to test p2p in isolation without running the entire pipeline inside the net crate

Reply:
We already have this as [`test_incoming_connect_with_single_geth`](https://github.com/paradigmxyz/reth/blob/27ed772d31b777882c6f1d1173b20609d3450624/crates/net/network/tests/it/connect.rs#L376) and a few other tests, although in those cases geth doesn't have any blocks. We could add to those with a test using the clique-configured geth, but am curious if you had anything in mind